reconcile/external_resources/aws.py

```diff
@@ -168,7 +169,10 @@ def resolve(
             and (target := blue_green_deployment.get("target"))
             and (parameter_group := target.get("parameter_group"))
         ):
-            target["parameter_group"] = rvr._get_values(parameter_group)
+            data["blue_green_deployment"] = copy.deepcopy(blue_green_deployment)
+            data["blue_green_deployment"]["target"]["parameter_group"] = (
+                rvr._get_values(parameter_group)
+            )
```

Comment:
for better performance, no need to do `deepcopy` since we know which field to change, use immutable deep merge

```python
            data["blue_green_deployment"] = blue_green_deployment | {
                "target": target | {
                    "parameter_group": rvr._get_values(parameter_group)
                }
            }
```